### PR TITLE
DI-539 Handle DoS Location Data

### PR DESCRIPTION
**Pull request by JackPlowman**
# Task Branch Pull Request

**<https://nhsd-jira.digital.nhs.uk/browse/DI-539>**

## Description of Changes

This PR takes the information from the location table and when a postcode change is required the information is applied to the services table of the dos db.

Also I have done some refactoring such as converting large functions into smaller functions in the compare_data.py file. As well the DoS logs are logged after the save has been applied to only log when a change has actually been made. This guarantees the change has been made before the log is sent to DoS Splunk.

## Type of change

- New feature (non-breaking change which adds functionality)
- Refactoring (non-breaking change which improves the structure of the code)

## Development Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the [code formatting checks](../README.md#code-quality)
- [x] I have run the [code quality checks](../README.md#code-quality)
- [x] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [x] I have updated Dependabot to include my changes (if applicable)

## Code Reviewer Checklist

- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I have checked any ignore commands for code linting tools and I agree that the code is safe

### CODE REVIEW

File: application/common/dos.py [hunk 1]
```diff
@@ -25,13 +25,18 @@ class DoSService:
...
     address: str
+    town: str
     postcode: str
     web: str
     typeid: int
     statusid: int
     publicphone: str
     publicname: str
     servicename: str
+    easting: int
```
Comment (rohith1122) [inline application/common/dos.py:36]: Attributes easting and northing hold only numbers, not decimals?

File: application/service_sync/dos_data.py [hunk 2]
```diff
@@ -149,10 +149,17 @@ def save_demographics_into_db(connection: connection, service_id: int, demograph
...
-            SQL(", ".join(f"{key} = '{value}'" for key, value in demographics_changes.items()))
+        logger.info(
+            f"Demographics changes found for service id {service_id}",
+            extra={"demographics_changes": demographics_changes},
         )
+        columns_and_values = [
+            SQL("{} = {}").format(Identifier(key), Literal(value)).as_string(connection)
+            for key, value in demographics_changes.items()
+        ]
+        print(columns_and_values)
```
Comment (rohith1122) [inline application/service_sync/dos_data.py:160]: could remove print statement

File: application/common/dos.py (hunk 1 above)
Comment (JackPlowman) [reply application/common/dos.py:36]: I've changed the type hints to floats to allow for decimal but as they are only type hints it shouldn't make a difference to the way the variables are used

File: application/service_sync/dos_data.py (hunk 2 above)
Comment (JackPlowman) [reply application/service_sync/dos_data.py]: I've removed this print statement used for debugging

**Review by rohith1122 — APPROVED** [review]
LGTM

**Pull request merged**